src/libraries/System.Transactions.Local/src/System/Transactions/TransactionManager.cs

```diff
@@ -310,6 +310,31 @@ public static TimeSpan DefaultTimeout
                 }
                 return s_defaultTimeout;
             }
+            set
+            {
+                TransactionsEtwProvider etwLog = TransactionsEtwProvider.Log;
+                if (etwLog.IsEnabled())
+                {
+                    etwLog.MethodEnter(TraceSourceType.TraceSourceBase, "TransactionManager.set_DefaultTimeout");
+                }
+
+                s_defaultTimeout = ValidateTimeout(value);
+
+                if (s_defaultTimeout != DefaultSettingsSection.Timeout)
+                {
+                    if (etwLog.IsEnabled())
+                    {
+                        etwLog.ConfiguredDefaultTimeoutAdjusted();
+                    }
+                }
+
+                s_defaultTimeoutValidated = true;
```

Comment:
Should DefaultTimeout always be less than MaxTimeout? If so, when setting either of the two property values, the other property value needs to be validated and adjusted accordingly.

Reply:
My intuition says yes, default should always be less than max. What did the config system do in 4.8?

It appears that ValidateTimeout() already handles this case here. But a similar validtation would need to be added in the setter for MaxTimeout.

Reply:
In 4.8 it did one-time loading from config settings and adjust, test result shows default always less than max. I have added ValidateTimeout for default timeout value in the MaxTimeout setter.